sources/sqlserver/infoschema.go

```diff
@@ -0,0 +1,383 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlserver
+
+import (
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/sources/common"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+type InfoSchemaImpl struct {
+	DbName string
+	Db     *sql.DB
+}
+
+// GetToDdl function below implement the common.InfoSchema interface.
+func (isi InfoSchemaImpl) GetToDdl() common.ToDdl {
+	return ToDdlImpl{}
+}
+
+// GetTableName returns table name.
+func (isi InfoSchemaImpl) GetTableName(schema string, tableName string) string {
+	if schema == "public" { // Drop 'public' prefix.
+		return tableName
+	}
+	return fmt.Sprintf("%s.%s", schema, tableName)
+}
+
+// ProcessDataRows performs data conversion for source database
+// 'db'. For each table, we extract data using a "SELECT *" query,
+// convert the data to Spanner data (based on the source and Spanner
+// schemas), and write it to Spanner.  If we can't get/process data
+// for a table, we skip that table and process the remaining tables.
+//
+// Note that the database/sql library has a somewhat complex model for
+// returning data from rows.Scan. Scalar values can be returned using
+// the native value used by the underlying driver (by passing
+// *interface{} to rows.Scan), or they can be converted to specific go
+// types.
+// We choose to do all type conversions explicitly ourselves so that
+// we can generate more targeted error messages: hence we pass
+// *interface{} parameters to row.Scan.
+func (isi InfoSchemaImpl) ProcessData(conv *internal.Conv, srcTable string, srcSchema schema.Table, spTable string, spCols []string, spSchema ddl.CreateTable) error {
+	rowsInterface, err := isi.GetRowsFromTable(conv, srcTable)
+	if err != nil {
+		conv.Unexpected(fmt.Sprintf("Couldn't get data for table %s : err = %s", srcTable, err))
+		return err
+	}
+	rows := rowsInterface.(*sql.Rows)
+	defer rows.Close()
+	srcCols, _ := rows.Columns()
+	v, scanArgs := buildVals(len(srcCols))
+	for rows.Next() {
+		// get RawBytes from data.
+		err := rows.Scan(scanArgs...)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Couldn't process sql data row: %s", err))
+			// Scan failed, so we don't have any data to add to bad rows.
+			conv.StatsAddBadRow(srcTable, conv.DataMode())
+			continue
+		}
+		values := valsToStrings(v)
+		ProcessDataRow(conv, srcTable, srcCols, srcSchema, spTable, spCols, spSchema, values)
+	}
+	return nil
+}
+
+// GetRowsFromTable returns a sql Rows object for a table.
+func (isi InfoSchemaImpl) GetRowsFromTable(conv *internal.Conv, srcTable string) (interface{}, error) {
+	q := fmt.Sprintf(`SELECT * FROM %s;`, srcTable)
```

Comment:
In SQL server can we  fire queries across databases ? If yes then this query needs to include database name

Reply:
Yes, queries across databases are possible in SQL Server. But here, as no database is passed, the query will be executed on the database which is in the connection object/context and which is supplied through the command line.

Any other use case that can cause any issue?

Reply:
If possible, its best we keep it consistent across all queries. Since we do specify the db name in other places, maybe we should do that here as well.

Reply:
Agreed, will update the queries.